p4-16/spec/P4-16-spec.mdk

```diff
@@ -4654,6 +4654,32 @@ bool b1 = (bit<32>)x == z;  // cast needed
 bool b2 = x == y;  // no cast needed
 ~ End P4Example
 
+## Operations on types that are type variables { #sec-type-variable-operations }
+
+Because functions, methods, control, and parsers can be generic,
+they offer the possibility of declaring values with types that
+are type variables:
+
+~ Begin P4Example
+control C<T>() {
+    apply {
+        T x;  // the type of x is T, a type variable
+    }
+}
+~ End P4Example
+
+The type of such objects is not known until the control is
+instantiated with specific type arguments.
+
+Currently the only operations that are available for such values are
+assignment (explicit through `=`, or implicit, through argument passing).
+This behavior is similar to languages such as Java, and different from
+languages such as C++.
+
+A future version of P4 may introduce a notion of _type constraints_
+which would enable more operations on such values.  Because of this
+limitaiton, currently such values of limited utility.
```

Comment:
suggested wordsmithing: "such values are currently of limited utility."

Reply:
There's a typo in last line: 'limitaiton' -> 'limitation'